packages/vite/src/node/config.ts

```diff
@@ -853,6 +853,8 @@ export async function loadConfigFromFile(
     if (!isTS && !isMjs) {
       // 1. try to directly require the module (assuming commonjs)
       try {
+        const bundled = await bundleConfigFile(resolvedPath)
+        dependencies = bundled.dependencies
         // clear cache in case of server restart
         delete require.cache[require.resolve(resolvedPath)]
         userConfig = require(resolvedPath)
```

Comment:
I feel the deps collection should be executed after the success required otherwise if there are syntax errors in the config will throw the error message from esbuild first and be less DX friendly to me.

Meanwhile, note that for now only the cache of root config has been cleared on requiring, meaning even if you collect the deps for HMR, it will still require the old nest module cache, which will make this only a partial fix. I'd feel it's better to always bundle and exclude the config file for consistency. (/cc https://github.com/egoist/bundle-require)

Reply:
This is the case, there will still be the problem of cache not being updated for nested dependencies. Therefore, the collection of dependencies is still needed in this way, but the configuration file is integrated into the bundle every time the dependency is updated. Does that mean so? (By the way, the dependency should be after require for DX, thanks for reminding)

Reply:
I'd like to always bundle and require the bundled code.This will solve the cache problem.

Reply:
hello?